sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/CosmosKeyCredential.java

```diff
@@ -9,17 +9,27 @@
  */
 public class CosmosKeyCredential {
 
-    private String masterKey;
+    private String key;
 
-    public CosmosKeyCredential(String masterKey) {
-        this.masterKey = masterKey;
+    //  Stores key's hashcode for performance improvements
+    private int keyHashCode;
+
+    public CosmosKeyCredential(String key) {
+        this.key = key;
+        this.keyHashCode = key.hashCode();
+    }
+
+    public String key() {
+        return key;
     }
 
-    public String getMasterKey() {
-        return masterKey;
+    public CosmosKeyCredential key(String key) {
+        this.key = key;
+        this.keyHashCode = key.hashCode();
+        return this;
     }
 
-    public void setMasterKey(String masterKey) {
-        this.masterKey = masterKey;
+    public int keyHashCode() {
```

Comment:
why is this public?

Reply:
This is used in BaseAuthorizationTokenProvider for performance optimization reasons.